app/com/arpnetworking/rollups/LastDataPointMessage.java

```diff
@@ -50,17 +58,21 @@ public RollupPeriod getPeriod() {
 
     private LastDataPointMessage(final Builder builder) {
         super(builder);
-        _metricName = builder._metricName;
+        _sourceMetricName = builder._sourceMetricName;
+        _rollupMetricName = builder._rollupMetricName;
         _tags = builder._tags;
-        _lastDataPointTime = builder._lastDataPointTime;
+        _sourceLastDataPointTime = builder._sourceLastDataPointTime;
+        _rollupLastDataPointTime = builder._rollupLastDataPointTime;
         _period = builder._period;
     }
 
-    private final String _metricName;
+    private final String _sourceMetricName;
+    private final String _rollupMetricName;
     private final RollupPeriod _period;
     private final ImmutableSet<String> _tags;
-    private final Instant _lastDataPointTime;
-    private static final long serialVersionUID = 5745882770658263619L;
+    private final Instant _sourceLastDataPointTime;
+    private final Instant _rollupLastDataPointTime;
+    private static final long serialVersionUID = 2800761302248621189L;
```

Comment:
Uh.... I have a cluster using rollups....

Reply:
I could revert the field rename which should keep it compatible but I think the addition of `sourceMetricName` will still be a problem because it cannot be null in a valid message.

I'll consider adding a new message as suggested.

Reply:
I duplicated/added a new message type. @vjkoskela and I had discussed this, worst-case the new/old clusters won't understand each other's messages but the job will be picked up on the next retry for the rollup. I opted for this instead of creating a delegating handler because it would not be possible for a new cluster to handle older messages since they will be missing required fields.